### PR TITLE
modified addContacts and deleteContacts to work both current and oponent

### PR DESCRIPTION
**Pull request by sziru**
finished #29

### CODE REVIEW

File: src/store/actions/contactsActions.js [hunk 1]
```diff
@@ -1,87 +1,132 @@
...
+        .collection("users")
+        .where("uid", "==", searchedEmail[0].uid)
+        .get()
+    ]).then(profiles => {
+      const profilesArray = [];
+      profiles.forEach(profile => {
+        profile.forEach(querySnapshot => {
+          const data = querySnapshot.data();
+          profilesArray.push(data);
+        });
```
Comment (iankameron) [inline src/store/actions/contactsActions.js]: 42-43 could be done in 1 line..?

profilesArray.push(querySnapshot.data());

File: src/store/actions/contactsActions.js [hunk 2]
```diff
@@ -1,87 +1,132 @@
...
+      });
+      const hasContactTwo = profilesArray[0].contacts.filter(contact => {
+        return contact.uid === profilesArray[1].uid;
+      });
+
+      if (hasContact.length === 0) {
+        let removedContact = { ...profilesArray[0] };
+        delete removedContact["contacts"];
+        profilesArray[1].contacts.push(removedContact);
+      }
```
Comment (iankameron) [inline src/store/actions/contactsActions.js]: "removedContact" stores the information of the person you're adding to the other persons contact.
I understand you then remove the contact data from it before you push it, but maybe a different name (like "newContactData") would be more logical?

File: src/store/actions/contactsActions.js [hunk 3]
```diff
@@ -1,87 +1,132 @@
...
+    console.log(userID, searchedEmail);
+    Promise.all([
+      firestore
+        .collection("users")
+        .where("uid", "==", userID)
+        .get(),
+      firestore
+        .collection("users")
+        .where("uid", "==", searchedEmail.uid)
+        .get()
```
Comment (iankameron) [inline src/store/actions/contactsActions.js]: you already have the whole user object in "searchedEmail"?

File: src/store/actions/contactsActions.js [hunk 4]
```diff
@@ -1,87 +1,132 @@
...
-    // .then((filteredArray) => 
-    // {
-    // })
-    .catch((err) => {
-      dispatch({ type: "DELETE_CONTACT_ERROR", err})
-    })
 
+      
+        const currentFilteredContects = profilesArray[0].contacts.filter(item => {return item.uid !== searchedEmail.uid})
+        firestore
```
Comment (iankameron) [inline src/store/actions/contactsActions.js]: please fix spelling of "Contacts"

File: src/store/actions/contactsActions.js [hunk 5]
```diff
@@ -1,87 +1,132 @@
...
+          .collection("users")
+          .doc(userID)
+          .update({
+            contacts: currentFilteredContects
+          })
+      
+        const oponentFilteredContects = profilesArray[1].contacts.filter(item => {return item.uid !== userID})
```
Comment (iankameron) [inline src/store/actions/contactsActions.js]: "opponent" is generally a term used for somebody that is working against you.
Please "otherUserFilteredContacts" or something that is easier to understand.

File: src/store/actions/contactsActions.js (hunk 3 above)
Comment (sziru) [reply src/store/actions/contactsActions.js]: searchedEmail -> searchedProfile

**Pull request merged**